pkg/apis/pravega/v1beta1/pravega.go

```diff
@@ -323,6 +323,24 @@ func (s *PravegaSpec) withDefaults() (changed bool) {
 		}
 	}
 
+	if s.SegmentStoreResources.Requests == nil {
+		changed = true
+		s.SegmentStoreResources.Requests = map[corev1.ResourceName]resource.Quantity{
```

Comment:
Are you sure at this point in the code that `s.SegmentStoreResources.Limits` exists?

Reply:
yes because in `ValidateSegmentStoreMemorySettings` method we are first checking whether `p.Spec.Pravega.SegmentStoreResources.Limits == nil` and in turn making sure user does pass Limits